### ISSUE TITLE
valgrind (3.13.0)

### ISSUE THREAD

**Issue by crbaird** [openhpc/ohpc#510]
http://valgrind.org/downloads/current.html

**Comment by koomie** [comment]
@pawosm-arm, could I ask you to comment on the applicability of the aarch64 valgrind patches to the latest valgrind release (v3.13.0)?  It looks like the majority of the patches landed in some form upstream.  There was one change in scheduler.c from rev16269.patch that looked to still apply so I created a subset patch for v3.13.0.  



```
Index: coregrind/m_scheduler/scheduler.c
===================================================================
--- coregrind/m_scheduler/scheduler.c   (revision 16266)
+++ coregrind/m_scheduler/scheduler.c   (revision 16269)
@@ -957,6 +957,10 @@
    vg_assert(VG_(in_generated_code) == True);
    VG_(in_generated_code) = False;
 
+#if defined(VGA_mips32) || defined(VGA_mips64)
+   tst->arch.vex.guest_LLaddr = (HWord)(-1);
+#endif
+
```

The code around the thunderx_always_use_fallback_LLSC patch looks to have changed a bit - I wasn't sure if it still needs to go in or not.

**Comment by pawosm-arm** [comment]
Ok, I see what happened. I created original LLSC fallback patch for AArch64, but, for the valgrind version that was provided by OpenHPC back then, one upstream patch (from SVN) was needed. And I copied it as-is, with all this mips part. Seems like things changed a bit upstream before 3.13.0 was released. Nowadays, this rev16269.v3.13.0.patch is not needed anymore.

Otherwise, everything needed for LLSC fallback is in the v3.13.0 including the VEX part.
